From 2f86d550564a404851c51ba6054d02dee2be08cf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Sat, 30 Jul 2011 22:53:33 +0800
Subject: [PATCH] Revert "Input: synaptics_i2c_rmi: Add irqflags to platform
 data struct to pass them to driver"

This reverts commit 252e0e32acd8fc3ff204844c51fdacf3e57ad001.
---
 drivers/input/touchscreen/synaptics_i2c_rmi.c | 5 +----
 include/linux/synaptics_i2c_rmi.h             | 1 -
 2 files changed, 1 insertion(+), 5 deletions(-)

diff --git a/drivers/input/touchscreen/synaptics_i2c_rmi.c b/drivers/input/touchscreen/synaptics_i2c_rmi.c
index f37c0c0fba4d..e003e42b6026 100644
--- a/drivers/input/touchscreen/synaptics_i2c_rmi.c
+++ b/drivers/input/touchscreen/synaptics_i2c_rmi.c
@@ -265,7 +265,6 @@ static int synaptics_ts_probe(
 	uint16_t max_x, max_y;
 	int fuzz_x, fuzz_y, fuzz_p, fuzz_w;
 	struct synaptics_i2c_rmi_platform_data *pdata;
-	unsigned long irqflags;
 	int inactive_area_left;
 	int inactive_area_right;
 	int inactive_area_top;
@@ -345,7 +344,6 @@ static int synaptics_ts_probe(
 			pdata++;
 		ts->flags = pdata->flags;
 		ts->sensitivity_adjust = pdata->sensitivity_adjust;
-		irqflags = pdata->irqflags;
 		inactive_area_left = pdata->inactive_left;
 		inactive_area_right = pdata->inactive_right;
 		inactive_area_top = pdata->inactive_top;
@@ -363,7 +361,6 @@ static int synaptics_ts_probe(
 		fuzz_p = pdata->fuzz_p;
 		fuzz_w = pdata->fuzz_w;
 	} else {
-		irqflags = 0;
 		inactive_area_left = 0;
 		inactive_area_right = 0;
 		inactive_area_top = 0;
@@ -515,7 +512,7 @@ static int synaptics_ts_probe(
 		goto err_input_register_device_failed;
 	}
 	if (client->irq) {
-		ret = request_irq(client->irq, synaptics_ts_irq_handler, irqflags, client->name, ts);
+		ret = request_irq(client->irq, synaptics_ts_irq_handler, 0, client->name, ts);
 		if (ret == 0) {
 			ret = i2c_smbus_write_byte_data(ts->client, 0xf1, 0x01); /* enable abs int */
 			if (ret)
diff --git a/include/linux/synaptics_i2c_rmi.h b/include/linux/synaptics_i2c_rmi.h
index 5539cc520779..0e40703807f4 100644
--- a/include/linux/synaptics_i2c_rmi.h
+++ b/include/linux/synaptics_i2c_rmi.h
@@ -32,7 +32,6 @@ struct synaptics_i2c_rmi_platform_data {
 				/* If non-zero another array entry follows */
 	int (*power)(int on);	/* Only valid in first array entry */
 	uint32_t flags;
-	unsigned long irqflags;
 	uint32_t inactive_left; /* 0x10000 = screen width */
 	uint32_t inactive_right; /* 0x10000 = screen width */
 	uint32_t inactive_top; /* 0x10000 = screen height */
-- 
2.35.3

